contracts/Validator.sol

```diff
@@ -0,0 +1,301 @@
+    /// @param _amount amount to stake.
+    /// @param _stakeRewards restake rewards.
+    function restake(
+        address _sender,
+        uint256 _validatorId,
+        uint256 _amount,
+        bool _stakeRewards
+    ) external override isOperator {
+        // get operator
+        INodeOperatorRegistry operator = getOperator();
+
+        // get stakeManager
+        address stakeManager = operator.getStakeManager();
+
+        // approve Polygon token to stake manager totalAmount
+        IERC20 polygonERC20 = IERC20(operator.getPolygonERC20());
+
+        // transfer tokens from user to this contract
+        polygonERC20.safeTransferFrom(_sender, address(this), _amount);
+
+        // approve to stakeManager
+        polygonERC20.safeApprove(stakeManager, _amount);
+
+        // call polygon stake manager
+        IStakeManager(stakeManager).restake(
+            _validatorId,
+            _amount,
+            _stakeRewards
+        );
+    }
+
+    /// @notice Unstake a validator from the Polygon stakeManager contract.
+    /// @dev Unstake a validator from the Polygon stakeManager contract by passing the validatorId
+    /// @param _validatorId validatorId.
+    function unstake(uint256 _validatorId) external override isOperator {
+        // call polygon stake manager
+        IStakeManager(getStakeManager()).unstake(_validatorId);
+    }
+
+    /// @notice Allows to top up heimdall fees.
+    /// @param _heimdallFee amount
+    function topUpForFee(address _sender, uint256 _heimdallFee)
+        external
+        override
+        isOperator
+    {
+        INodeOperatorRegistry operator = getOperator();
+
+        // get stakeManager
+        address stakeManager = operator.getStakeManager();
+
+        IERC20 polygonERC20 = IERC20(operator.getPolygonERC20());
+
+        // transfer tokens from user to this contract
+        polygonERC20.safeTransferFrom(_sender, address(this), _heimdallFee);
+
+        // approve to stakeManager
+        polygonERC20.safeApprove(stakeManager, _heimdallFee);
+
+        // call polygon stake manager
+        IStakeManager(stakeManager).topUpForFee(address(this), _heimdallFee);
+    }
+
+    /// @notice Allows to withdraw rewards from the validator.
+    /// @dev Allows to withdraw rewards from the validator using the _validatorId. Only the
+    /// owner can request withdraw in this the owner is this contract.
+    /// @param _validatorId validator id.
+    function withdrawRewards(uint256 _validatorId)
+        external
+        override
+        isOperator
+        returns (uint256)
+    {
+        INodeOperatorRegistry operator = getOperator();
+
+        // call polygon stake manager
+        // withdraw rewards
+        IStakeManager(operator.getStakeManager()).withdrawRewards(_validatorId);
+
+        // transfer rewards to lido contract.
+        IERC20 polygonERC20 = IERC20(operator.getPolygonERC20());
+        uint256 balance = polygonERC20.balanceOf(address(this));
```

Comment:
No, each validator has one owner, and the only way to stake is using the operator contract. Here I transfer the reward balance to the lido contract.  But for now, we are not going to use it because the rewards are too small.